### ISSUE TITLE
[Feature] Test collector not recognizing Symbolic Link

### ISSUE THREAD

**Issue by sanjoyghosh** [microsoft/playwright#17522]
I am packaging a bunch of Playwright Test Scripts as symbolic links to the original location.

In the new package, all the test scripts in the "tests" folder, i.e. tests/*.spec.ts files are symbolic links to *.spec.ts files elsewhere.  And it doesn't recognize any tests to run.

At line 793, https://github.com/microsoft/playwright/blob/main/packages/playwright-test/src/runner.ts#L793
It only checks if the given entryPath is a file.  This misses entryPath's that are symbolic links that resolve to a regular file.

Basically, the addition of the following line fixes this problem:
```
else if (entry.isSymbolicLink() && fs.lstatSync(fs.realpathSync(entryPath)).isFile() && entryStatus === 'included') 
    files.push(entryPath);
```

**Comment by pavelfeldman** [comment]
I'm not sure this will be compatible with Playwright. We are going to resolve the link, use the absolute path and then build a relative path from the test dir to that absolute path for the sake of reporting, snapshots and artifact collection. That latter part will obviously fail.

**Comment by sanjoyghosh** [comment]
Thanks for the quick response Pavel.  Not sure I understand.

In the suggested code, I still add the symbolic link to the list of test files, not the resolved path.  So I don't understand why anything should fail.  Unless I am missing something.

**Comment by pavelfeldman** [comment]
Playwright relies on the fact that all the resolved test files are under the root testDir for the reasons I listed above. Breaking this assumption would be hard.

**Comment by sanjoyghosh** [comment]
Thanks Pavel for settling that.  I will make sure to copy the files instead of using symbolic links.
